File: leetcode/Python/convert-bst-to-greater-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def convertBST(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        s = [0]
        def traverse(node):
            if node is None:
                return
            traverse(node.right)
            node.val += s[0]
            s[0] = node.val
            traverse(node.left)
        traverse(root)
        return root

File: leetcode/C++/convert-bst-to-greater-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void traverse(TreeNode* node, int &sum){
        if(node==nullptr) return;
        traverse(node->right, sum);
        node->val+=sum;
        sum=node->val;
        traverse(node->left, sum);
    }
    TreeNode* convertBST(TreeNode* root) {
        int sum=0;
        traverse(root, sum);
        return root;
    }
};